dbt/include/trino/macros/adapters.sql

```diff
@@ -4,33 +4,25 @@
 -- - get_columns_in_relation
 
 {% macro trino__get_columns_in_relation(relation) -%}
-  {% call statement('get_columns_in_relation', fetch_result=True) %}
-      select
-          column_name,
-          case when regexp_like(data_type, 'varchar\(\d+\)') then 'varchar'
-               else data_type
-          end as data_type,
-          case when regexp_like(data_type, 'varchar\(\d+\)') then
-                  from_base(regexp_extract(data_type, 'varchar\((\d+)\)', 1), 10)
-               else NULL
-          end as character_maximum_length,
-          NULL as numeric_precision,
-          NULL as numeric_scale
-
-      from
-      {{ relation.information_schema('columns') }}
-
-      where
-        table_name = '{{ relation.name }}'
-        {% if relation.schema %}
-        and table_schema = '{{ relation.schema | lower }}'
-        {% endif %}
-      order by ordinal_position
-
-  {% endcall %}
+  {%- set sql -%}
+    describe {{ relation }}
+  {%- endset -%}
+  {%- set result = run_query(sql) -%}
+
+  {% set maximum = 10000 %}
+  {% if (result | length) >= maximum %}
+    {% set msg %}
+      Too many columns in relation {{ relation }}! dbt can only get
```

Comment:
Remove exclamation mark.

Reply:
This is coming straight from the dbt-snowflake implementation. I would personally not change this. It seems already exceptionally high to have a table with 10000 columns.